Thanks @KrishnaSai2020. This doctest is still failing in the file, do you want to fix it here too?

`FAILED pandas/io/formats/format.py::pandas.io.formats.format.EngFormatter.__call__`

--- a/pandas/io/formats/format.py
+++ b/pandas/io/formats/format.py
@@ -1958,7 +1958,7 @@ class EngFormatter:
         representing the power of 1000 of the original number. Some examples:
         >>> format_eng = EngFormatter()
         >>> format_eng(0)       # for self.accuracy = 0
-        ' 0'
+        '0'
 
         >>> format_eng(1000000) # for self.accuracy = 1,
                                 #     self.use_eng_prefix = True

--- a/pandas/io/formats/format.py
+++ b/pandas/io/formats/format.py
@@ -1952,7 +1952,7 @@ class EngFormatter:
         self.accuracy = accuracy
         self.use_eng_prefix = use_eng_prefix
 
-    def __call__(self, num: int | float) -> str:
+    def __call__(self, num: int | float) -> int:
         """
         Formats a number in engineering notation, appending a letter
         representing the power of 1000 of the original number. Some examples:
@@ -1983,6 +1983,9 @@ class EngFormatter:
             return "inf"
 
         sign = 1
+        if dnum == 0:
+            formatted = 0
+            return formatted
 
         if dnum < 0:  # pragma: no cover
             sign = -1

--- a/pandas/io/formats/format.py
+++ b/pandas/io/formats/format.py
@@ -1952,13 +1952,13 @@ class EngFormatter:
         self.accuracy = accuracy
         self.use_eng_prefix = use_eng_prefix
 
-    def __call__(self, num: int | float) -> int:
+    def __call__(self, num: int | float) ->str:
         """
         Formats a number in engineering notation, appending a letter
         representing the power of 1000 of the original number. Some examples:
         >>> format_eng = EngFormatter()
         >>> format_eng(0)       # for self.accuracy = 0
-        '0'
+        ' 0'
 
         >>> format_eng(1000000) # for self.accuracy = 1,
                                 #     self.use_eng_prefix = True
@@ -1984,7 +1984,7 @@ class EngFormatter:
 
         sign = 1
         if dnum == 0:
-            formatted = 0
+            formatted = ' 0'
             return formatted
 
         if dnum < 0:  # pragma: no cover

